### PR TITLE
Moiz-WFLOW-2577-Update Sitemap Generation logic

### PR DESCRIPTION
**Pull request by moiz-deriv**
fix: updated sitemap generation to remove specific url from the sitemap which were highlighted by the SEO team

### CODE REVIEW

File: .github/modify_sitemap.js [hunk 1]
```diff
@@ -45,13 +106,27 @@ fs.readFile(inputFile, "utf8", (err, data) => {
     if (locPattern.test(match) && !locDirectPattern.test(match)) {
       return ""; // Remove entire <url> block if <loc> contains an unwanted segment
     }
+    match.replace(linkPattern, (link) => {
+      console.log("Checking Link", link);
```
Comment (matin-deriv) [inline .github/modify_sitemap.js:110]: why do we need a console in production, please remove.

File: .github/modify_sitemap.js [hunk 2]
```diff
@@ -34,9 +33,71 @@ fs.readFile(inputFile, "utf8", (err, data) => {
   const urlBlockPattern = /(<url>[\s\S]*?<\/url>)/g;
   const xhtmlLinkPattern =
     /(<xhtml:link[^>]*?href="https:\/\/[^\/]+\/[^\/]+\/locations[^"]*"[^>]*?>)/g;
+  const linkPattern =
+    /(<xhtml:link[^>]*?href="https:\/\/[^\/]+\/[^\/]+[^"]*"[^>]*?>)/g;
   const locPattern = new RegExp(`https://${newDomain}\/[^\/]+\/locations`, "i"); // Matches URLs with an extra segment before /locations
   const locDirectPattern = new RegExp(`https://${newDomain}\/locations\/`, "i"); // Matches URLs directly starting with /locations
 
+  const staticDocUrls = [
```
Comment (matin-deriv) [inline .github/modify_sitemap.js:41]: Best to dynamically define the urls to make it future proof.

`
const languages = [
  "sw", "bn", "ko", "pl", "tr", "it", "fr", "th", "zh-cn", "si", 
  "vi", "zh-tw", "es", "pt", "ar", "de", "ru"
];

const baseUrl = "https://deriv.com";
const endpoints = [
  "/blog/posts/what-are-interest-rates-and-why-do-they-matter-in-trading",
  "/blog/posts/why-beginners-need-to-have-an-online-trading-demo-account",
  "/blog/posts/the-truth-is-out-5-copy-trading-myths-debunked",
  "/blog/posts/trading-with-crypto-top-3-myths",
  "/blog/posts/eur-usd-holds-strong-before-ecb-meeting",
  "/blog/posts/eu-retail-sales-germany-inflation-rate",
  "/blog/posts/what-are-accumulator-options-a-beginners-guide-to-compounding-growth",
  "/signup"
];

// Generating all the URLs dynamically
const staticDocs = [];

languages.forEach(lang => {
  endpoints.forEach(endpoint => {
    staticDocs.push(`${baseUrl}/${lang}${endpoint}`);
  });
});

// Adding URLs without language prefixes if needed
endpoints.forEach(endpoint => {
  staticDocs.push(`${baseUrl}${endpoint}`);
});
`

Comment (moiz-deriv) [reply .github/modify_sitemap.js:41]: these are hyper specific scenerios where we donot want to remove for all languages neither for other domains

File: .github/modify_sitemap.js (hunk 1 above)
Comment (moiz-deriv) [reply .github/modify_sitemap.js]: removed

**Review by matin-deriv — APPROVED** [review]

**Pull request merged**